readthedocs/search/parsers.py

```diff
@@ -140,21 +140,68 @@ def _parse_sections(self, title, body):
         except Exception as e:
             log.info('Unable to index section', section=str(e))
 
+        # All terms in dls are treated as sections.
+        # After indexing them, we remove them from the body such that their contents aren't
+        # repeated in the following indexing
+        dls = body.css("dl")
+        dl_nodes_to_be_removed = []
+        for dl in dls:
+            # Select all dts with id defined
+            dts = dl.css('dt[id]:not([id=""])')
+
+            # https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dt
+            # multiple <dt> elements in a row indicate several terms that are
+            # all defined by the immediate next <dd> element.
+            for dt in dts:
+                if dl not in dl_nodes_to_be_removed:
+                    dl_nodes_to_be_removed.append(dl)
+                try:
+                    title, _id = self._parse_dt(dt)
+                    # https://developer.mozilla.org/en-US/docs/Web/CSS/General_sibling_combinator
+                    dd = dt.css_first("dt ~ dd")
+                    # We only index a dt with an id attribute and an accompanying dd
+                    if not dd or not _id:
+                        continue
+
+                    # The content of the <dt> section is the content of the accompanying <dd>
+                    content = dd.text()
+                    yield {
+                        "id": _id,
+                        "title": title,
+                        "content": content,
+                    }
+                except Exception as e:
+                    log.info("Unable to index dt section.", section=str(e))
+
+        # Remove the <dl> nodes that were found to have indexed content
+        for node in dl_nodes_to_be_removed:
+            node.decompose()
```

Comment:
@stsewd I suspect that you didn't see this? This is where indexed `<dl>` nodes are removed.

Reply:
This is now handled based on some assumptions. I added https://requests.readthedocs.io/en/latest/api/#requests.Response as raw test data so we can track any changes in behavior closely. The test runs very fast, so I'm not concerned about the length of this HTML file.